sqldb/sqldb.go

```diff
@@ -0,0 +1,72 @@
+package sqldb
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"time"
+
+	"github.com/jackc/pgx/v4"
+	"github.com/lightningnetwork/lnd/sqldb/sqlc"
+)
+
+var (
+	// DefaultStoreTimeout is the default timeout used for any interaction
```

Comment:
question: are these 10 sec common practice, would be good to have a more detailed comment why we choose 10sec here?

Reply:
10s sounded like a good upper bound for how long the "average" query can be running for before failing. For queries where more time is expected (reporting is an example) we can use a custom timeout.

Happy to use another number here, but it is just "how long should we wait for a query or for all the queries in a tx before failing"